src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -270,7 +270,7 @@ private ValueTask<HttpConnection> GetOrReserveHttp11ConnectionAsync(Cancellation
                 }
 
                 HttpConnection conn = cachedConnection._connection;
-                if (cachedConnection.IsUsable(now, pooledConnectionLifetime, pooledConnectionIdleTimeout) &&
+                if (cachedConnection.IsUsable(now, pooledConnectionLifetime, Timeout.InfiniteTimeSpan) &&
```

Comment:
I wonder if we should just explicitly do the lifetime check here. IsUsable is doing three different checks (lifetime, idle, poll) and now we're skipping two of them here. Might be simpler to just move the specific lifetime check here. I think we already manually check lifetime in ReturnConnectionToPool anyway.

Reply:
Fine with me. I can always re-factor it to small function and call it from IsUsable() as needed.

Reply:
Actually it's in ReturnConnection, not ReturnConnectionToPool.

Reply:
Sounds good. If we refactor, we may want to call that from ReturnConnection as well, for consistency.